gitlab4j-api/src/main/java/org/gitlab4j/api/RepositoryApi.java

```diff
@@ -942,6 +931,38 @@ public CompareResults compare(Object projectIdOrPath, String from, String to) th
         return (compare(projectIdOrPath, from, to, false));
     }
 
+    /**
+     * Compare branches, tags or commits. This can be accessed without authentication
+     * if the repository is publicly accessible.
+     *
+     * @param projectIdOrPath the project in the form of an Long(ID), String(path), or Project instance
+     * @param from the commit SHA or branch name
+     * @param to the commit SHA or branch name
+     * @param fromProjectId The ID to compare from.
+     * @param straight specifies the comparison method, true for direct comparison between from and to (from..to),
+     *          false to compare using merge base (from…to)
+     * @return a CompareResults containing the results of the comparison
+     * @throws GitLabApiException if any exception occurs
+     */
+    public CompareResults compare(Object projectIdOrPath, String from, String to, Object fromProjectId, boolean straight)
```

Comment:
* `straight` should be a Boolean.
* `fromProjectId` should be a Long

```suggestion
    public CompareResults compare(Object projectIdOrPath, String from, String to, Long fromProjectId, Boolean straight)
```